interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -1330,6 +1333,67 @@ public void perRpcCreds(String jsonKey, InputStream credentialsStream, String oa
     oauth2AuthToken(jsonKey, credentialsStream, oauthScope);
   }
 
+  /** Sends a cacheable unary rpc using GET. Requires that the server is behind a caching proxy. */
+  public void cacheableUnary() {
+    // Build method descriptor manually to set idempotent and safe to true.
+    io.grpc.MethodDescriptor<
+            io.grpc.testing.integration.Messages.SimpleRequest,
+            io.grpc.testing.integration.Messages.SimpleResponse>
+        safeCacheableUnaryCallMethod =
+            io.grpc.MethodDescriptor
+                .<io.grpc.testing.integration.Messages.SimpleRequest,
+                    io.grpc.testing.integration.Messages.SimpleResponse>
+                    newBuilder()
+                .setType(io.grpc.MethodDescriptor.MethodType.UNARY)
+                .setFullMethodName(
+                    MethodDescriptor.generateFullMethodName(
+                        "grpc.testing.TestService", "CacheableUnaryCall"))
+                .setRequestMarshaller(
+                    io.grpc.protobuf.ProtoUtils.marshaller(
+                        io.grpc.testing.integration.Messages.SimpleRequest.getDefaultInstance()))
+                .setResponseMarshaller(
+                    io.grpc.protobuf.ProtoUtils.marshaller(
+                        io.grpc.testing.integration.Messages.SimpleResponse.getDefaultInstance()))
+                .setIdempotent(true)
+                .setSafe(true)
+                .build();
+    // Set fake user IP since some proxies (GFE) won't cache requests from localhost.
+    Metadata.Key<String> userIpKey = Metadata.Key.of("x-user-ip", Metadata.ASCII_STRING_MARSHALLER);
+    Metadata metadata = new Metadata();
+    metadata.put(userIpKey, "1.2.3.4");
+    Channel channelWithUserIpKey =
+        ClientInterceptors.intercept(channel, MetadataUtils.newAttachHeadersInterceptor(metadata));
+    SimpleRequest requests1And2 =
+        SimpleRequest.newBuilder()
+            .setResponseSize(1)
```

Comment:
Why is this requesting response details if it is just going to be thrown away? It doesn't say this should be sent in the interop spec.